platform/linux-generic/arch/aarch64/odp/api/abi/atomic_inlines.h

```diff
@@ -152,7 +152,62 @@ static inline int _odp_atomic_cas_acq_rel_u128(odp_atomic_u128_t *atom, odp_u128
 
 	return 0;
 }
-#else /* _ODP_LOCK_FREE_128BIT_ATOMICS */
+
+static inline void _odp_atomic_add_u32(odp_atomic_u32_t *atom, uint32_t val)
+{
+	__asm__ volatile("stadd   %w0, [%1]"
+			 :
+			 : "r" (val), "r" (atom)
+			 : "memory");
+}
+
+static inline void _odp_atomic_sub_u32(odp_atomic_u32_t *atom, uint32_t val)
+{
+	__asm__ volatile("neg     %w0, %w0\n"
+			 "stadd   %w0, [%1]"
+			 :
+			 : "r" (val), "r" (atom)
+			 : "memory");
```

Comment:
This code doesn't seem to work if optimization is enabled (`-O2`).

E.g. atomic validation test
```
Num of threads used = 12
	ODP API version: 1.31.0
	ODP implementation name:    odp-linux
	ODP implementation version: odp-linux 1.31.0-0 (v1.31.0) 1.31.0.0


     CUnit - A unit testing framework for C - Version 2.1-3
     http://cunit.sourceforge.net/


Suite: atomic
  Test: atomic_test_atomic_inc_dec ...FAILED
    1. atomic.c:696  - U32_INIT_VAL == odp_atomic_load_u32(&global_mem->a32u)
    2. atomic.c:697  - U64_INIT_VAL == odp_atomic_load_u64(&global_mem->a64u)
  Test: atomic_test_atomic_add_sub ...threads.c:212:wait_pthread():Bad exit status cpu #12 0x2
error: odph_thread_join() failed.
FAILED
    1. atomic.c:696  - U32_INIT_VAL == odp_atomic_load_u32(&global_mem->a32u)
    2. atomic.c:697  - U64_INIT_VAL == odp_atomic_load_u64(&global_mem->a64u)
  Test: atomic_test_atomic_fetch_inc_dec ...threads.c:212:wait_pthread():Bad exit status cpu #12 0x4
error: odph_thread_join() failed.
passed
  Test: atomic_test_atomic_fetch_add_sub ...threads.c:212:wait_pthread():Bad exit status cpu #12 0x4
error: odph_thread_join() failed.
passed
  Test: atomic_test_atomic_max_min ...threads.c:212:wait_pthread():Bad exit status cpu #12 0x4
error: odph_thread_join() failed.
passed
  Test: atomic_test_atomic_cas_inc_dec ...threads.c:212:wait_pthread():Bad exit status cpu #12 0x4
error: odph_thread_join() failed.
passed
  Test: atomic_test_atomic_xchg ...^C
```
I'm not that familiar with assembly coding, but I guess you shouldn't write to an input register.